daml/Tests/ExchangeTrade.daml

```diff
@@ -13,27 +13,28 @@ import DA.List
 
 setup : Script ()
 setup = do
-  cr@CommonContracts{..} <- onboardCommon
+  providers@Providers{..} <- onboardProviders
+  Assets{..} <- onboardAssets providers
```

Comment:
Again, avoid abusing `..`, except for in succinct, local scope! It makes the code much less legible.